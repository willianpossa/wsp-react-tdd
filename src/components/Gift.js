import React, { Component } from 'react';
import { Form, Button } from 'react-bootstrap'

class Gift extends Component {

    state = {
        person: '',
        gift: ''
    }

    // handleChange = ({ target: { value, name }}) => {
    //     this.setState({
    //         [name]: value
    //     });
    // }
    
    render() {
        const { person, gift } = this.state

        return (
            <div>
                <Form>
                    <Form.Group>
                        <Form.Label>Person</Form.Label>
                        <Form.Control 
                            type="text" 
                            className="input-person" 
                            value={ person }
                            onChange={ ({ target: { value }}) => {
                                this.setState(prevState => ({
                                    ...prevState,
                                    person: value
                                }))
                            } } 
                        />
                    </Form.Group>
                </Form>
            </div>
        )
    }
}

export default Gift;