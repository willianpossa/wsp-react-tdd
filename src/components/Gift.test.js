import React from 'react';
import { shallow } from 'enzyme';

import Gift from './Gift';

describe('Gift Component Tests', () => {
    const gift = shallow(<Gift />);

    it('renders correctly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('check if state has been initializes', () => {
        expect(gift.state()).toEqual({
            person: '',
            gift: ''
        })
    });

    describe('when user typing in `person` input', () => {
        const person = 'Willian';

        beforeEach(() => {
            gift.find('.input-person').simulate('change', { target: { value: person } });
        });

        it('update person `state`', () => {
            expect(gift.state().person).toEqual(person);
        });
    });
});

