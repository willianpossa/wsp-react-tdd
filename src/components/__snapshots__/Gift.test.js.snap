// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Component Tests renders correctly 1`] = `ShallowWrapper {}`;
